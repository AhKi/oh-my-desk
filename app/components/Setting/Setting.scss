.Setting {
  width: 100%;
  height: 100%;
}

.Setting__content {
  margin: 20px;
  padding: 10px;
  background-color: #f2f2f2;
  border: 1px solid #e1e1e1;
  border-radius: 4px;
}
