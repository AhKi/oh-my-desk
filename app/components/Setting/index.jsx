import React from 'react';

class Setting extends React.Component {
	render() {
		return (
			<div>
				Setting
			</div>
		);
	}
}

export default Setting;