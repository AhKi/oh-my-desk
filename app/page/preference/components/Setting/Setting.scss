.Setting {
  margin: 15px;
}

.Setting__title {
  font-weight: 600;
  font-size: 14px;
}

.Setting__list {
  display: flex;
  padding: 10px;
  background: white;
  font-size: 14px;
}

.Setting__list-set {
  box-shadow: 1px 2px rgba(0, 0, 0, 0.2);
}
