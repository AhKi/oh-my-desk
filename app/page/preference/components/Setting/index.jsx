import React from 'react';

const propTypes = {};
const defaultProps = {};

class Setting extends React.Component {
  render() {
    return (
      <div>Setting</div>
    );
  }
}

Setting.propTypes = propTypes;
Setting.defaultProps = defaultProps;

export default Setting;
