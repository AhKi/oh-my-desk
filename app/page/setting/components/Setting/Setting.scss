.Setting {
  width: 100%;
  height: 100%;
}

.Setting__content {
  margin: 20px;
  padding: 10px;
  background-color: $color-black;
  border-radius: 4px;
}
