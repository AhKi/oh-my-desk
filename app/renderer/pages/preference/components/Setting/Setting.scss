.Setting {
  margin: 0 15px 15px 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.Setting__title {
  font-weight: 600;
  font-size: 14px;
  margin: 10px 0;
}

.Setting__list {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  background: white;
  font-size: 14px;
  border-bottom: 1px solid $color-cool-gray200;
  height: 45px;
}

.Setting__list-set {
  box-shadow: 1px 2px rgba(0, 0, 0, 0.2);
}

.Setting__select {
  width: 400px;
}
