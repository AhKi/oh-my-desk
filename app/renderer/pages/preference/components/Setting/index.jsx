import React from 'react';
import PropTypes from 'prop-types';
import i18n from 'constants/i18n';
import ToggleButton from 'renderer/components/ToggleButton';
import WidgetModeConfirm from '../Modal/WidgetModeConfirm';
import ConfigureHotKeyContainer from '../../containers/ConfigureHotKeyContainer';
import './Setting.scss';

const propTypes = {
  defaultUserAgent: PropTypes.string,
  language: PropTypes.string,
  isOpenWidgetWhenStart: PropTypes.bool,
  isLaunchAppWhenLogin: PropTypes.bool,
  onModalOpen: PropTypes.func,
  onSetLanguageEnglish: PropTypes.func,
  onSetLanguageKorean: PropTypes.func,
  onToggleOpenWidgetWhenStart: PropTypes.func,
  onToggleOpenAppWhenLogin: PropTypes.func,
  onToggleWidgetDefaultUserAgent: PropTypes.func,
};

const defaultProps = {
  defaultUserAgent: 'DESKTOP',
  language: 'English',
  isOpenWidgetWhenStart: true,
  isLaunchAppWhenLogin: true,
  onModalOpen() {},
  onSetLanguageEnglish() {},
  onSetLanguageKorean() {},
  onToggleOpenWidgetWhenStart() {},
  onToggleOpenAppWhenLogin() {},
  onToggleWidgetDefaultUserAgent() {},
};

class Setting extends React.Component {
  constructor(props) {
    super(props);
    this.handleChangeLanguage = this.handleChangeLanguage.bind(this);
    this.handleOpenWidgetModeModal = this.handleOpenWidgetModeModal.bind(this);
  }

  handleChangeLanguage(e) {
    const language = e.target.value;
    const { onSetLanguageEnglish, onSetLanguageKorean } = this.props;

    if (language === 'English') {
      onSetLanguageEnglish();
    } else {
      onSetLanguageKorean();
    }
  }

  handleOpenWidgetModeModal() {
    const { onModalOpen, onToggleWidgetDefaultUserAgent } = this.props;

    onModalOpen(WidgetModeConfirm, {
      onChangeMode: onToggleWidgetDefaultUserAgent,
    });
  }

  render() {
    const text = i18n().preference;
    const {
      defaultUserAgent,
      language,
      isOpenWidgetWhenStart,
      isLaunchAppWhenLogin,
      onToggleOpenWidgetWhenStart,
      onToggleOpenAppWhenLogin,
    } = this.props;

    return (
      <div className="Setting">
        <div>
          <p className="Setting__title">{text.user}</p>
          <ul className="Setting__list-set">
            <li className="Setting__list">
              {text.autoStart}
              <ToggleButton
                isCheck={isLaunchAppWhenLogin}
                onToggle={onToggleOpenAppWhenLogin}
              />
            </li>
            <li className="Setting__list">
              {text.autoWidgetActive}
              <ToggleButton
                isCheck={isOpenWidgetWhenStart}
                onToggle={onToggleOpenWidgetWhenStart}
              />
            </li>
            <li className="Setting__list">
              {text.language}
              <select
                className="InputSet__select"
                value={language}
                onChange={this.handleChangeLanguage}
              >
                <option value="English">English</option>
                <option value="Korean">한국어</option>
              </select>
            </li>
          </ul>
        </div>
        <div>
          <p className="Setting__title">{text.shape}</p>
          <ul className="Setting__list-set">
            <li className="Setting__list">
              {text.defaultWidgetMode}
              <select
                className="InputSet__select"
                value={defaultUserAgent}
                onChange={this.handleOpenWidgetModeModal}
              >
                <option value="DESKTOP">{text.desktopMode}</option>
                <option value="MOBILE">{text.mobileMode}</option>
              </select>
            </li>
            <li className="Setting__list">
              {text.hotKeySearch}
              <ConfigureHotKeyContainer />
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

Setting.propTypes = propTypes;
Setting.defaultProps = defaultProps;

export default Setting;
