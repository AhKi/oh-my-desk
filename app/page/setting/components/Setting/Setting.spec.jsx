import React from 'react';
import { shallow } from 'enzyme';

import Setting from './';

describe('<Setting />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<Setting />);

    expect(wrapper).toMatchSnapshot();
  });
});
