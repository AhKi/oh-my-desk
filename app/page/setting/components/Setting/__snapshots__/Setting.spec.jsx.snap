// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Setting /> should match to snapshot when render default 1`] = `
<div
  className="Setting"
>
  <div
    className="Setting__content"
  >
    <h4>
      Setting
    </h4>
    Comming soon
  </div>
</div>
`;
